Lib/ufo2ft/featureWriters/kernFeatureWriter.py

```diff
@@ -628,15 +628,18 @@ def _makeKerningLookup(self, name, ignoreMarks=True):
                 )
         return lookup
 
-    def knownScriptsPerCodepoint(self, uv: int) -> set[str]:
+    def knownScriptsPerCodepoint(self, uv: int) -> Union[set[str], None]:
         if not self.context.knownScripts:
             # If there are no languagesystems and nothing to derive from Unicode
             # codepoints, consider everything common; it'll all end in DFLT/dflt
             # anyway.
             return {COMMON_SCRIPT}
         else:
             script_extension = unicodeScriptExtensions(uv)
-            return script_extension & (self.context.knownScripts | DFLT_SCRIPTS)
+            known = script_extension & (self.context.knownScripts | DFLT_SCRIPTS)
+            if len(known) > 0:
+                return known
+            return None
```

Comment:
if you like, this is equivalent

```suggestion
            return script_extension & (self.context.knownScripts | DFLT_SCRIPTS) or None
```